### PR TITLE
Add support to share post by email

### PR DESCRIPTION
**Pull request by pravintp**

### CODE REVIEW

File: mysite/blog/views.py [hunk 1]
```diff
@@ -22,3 +24,25 @@ def post_detail(request, year, month, day, post):
...
+                f"Read {post.title} at {post_url}\n\n"
+                f"{cd['name']}'s comments: {cd['comments']}"
+            )
+            send_mail(subject, message, "admin@myblog.com", [cd["to"]])
+            sent = True
+    else:
+        form = EmailPostForm()
+    return render(
+        request, "blog/post/share.html", {"post": post, "form": form, "sent": sent}
+    )
```
Comment (waranlogesh) [inline mysite/blog/views.py]: Not clear on what this is doing. Please clean it.

File: mysite/blog/templates/blog/post/share.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{% extends "blog/base.html" %}
```
Comment (waranlogesh) [inline mysite/blog/templates/blog/post/share.html:1]: indent

**Review by waranlogesh — CHANGES_REQUESTED** [review]

File: mysite/blog/views.py (hunk 1 above)
Comment (pravintp) [reply mysite/blog/views.py]: ok

File: mysite/blog/views.py [hunk 3]
```diff
@@ -1 +1,48 @@
...
+        form = EmailPostForm(request.POST)
+        if form.is_valid():
+            cd = form.cleaned_data
+            post_url = request.build_absolute_uri(post.get_absolute_url())
+            subject = f"{cd['name']} recommends you read " f"{post.title}"
+            message = (
+                f"Read {post.title} at {post_url}\n\n"
+                f"{cd['name']}'s comments: {cd['comments']}"
+            )
+            send_mail(subject, message, "admin@myblog.com", [cd["to"]])
```
Comment (waranlogesh) [inline mysite/blog/views.py]: Extract method to `share_post_by_email(post, email, name)`

File: mysite/blog/views.py [hunk 4]
```diff
@@ -1 +1,48 @@
...
+        publish__year=year,
+        publish__month=month,
+        publish__day=day,
+    )
+    return render(request, "blog/post/detail.html", {"post": post})
+
+
+def post_share(request, post_id):
+    post = get_object_or_404(Post, id=post_id, status="published")
+    sent = False
```
Comment (waranlogesh) [inline mysite/blog/views.py]: This is should be handled by form.errors

File: mysite/blog/templates/blog/post/share.html [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{% extends "blog/base.html" %}
+
+{% block title %}Share a post{% endblock %}
+{% block content %}
+ {% if sent %}
+ <h1>E-mail successfully sent</h1>
+ <p>
+ "{{ post.title }}" was successfully sent to {{ form.cleaned_data.to }}.
+ </p>
+ {% else %}
```
Comment (waranlogesh) [inline mysite/blog/templates/blog/post/share.html]: use messages

File: mysite/blog/templates/pagination.html [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
- {% if page.has_next %}
- <a href="?page={{ page.next_page_number }}">Next</a>
- {% endif %}
+   {% if page.has_next %}
+     <a href="?page={{ page.next_page_number }}">Next</a>
+   {% endif %}
  </span>
-</div>
\ No newline at end of file
+</div>
```
Comment (waranlogesh) [inline mysite/blog/templates/pagination.html]: This should not be part of this PR. Push is formating change directly to master.

File: mysite/blog/urls.py [hunk 7]
```diff
@@ -6,10 +6,11 @@
 app_name = "blog"
 
 urlpatterns = [
-    path("", views.post_list, name="post_list"),
+    path("", views.PostListView.as_view(), name="post_list"),
```
Comment (waranlogesh) [inline mysite/blog/urls.py]: This should not be part of this PR.

File: mysite/blog/forms.py [hunk 8]
```diff
@@ -0,0 +1,8 @@
+from django import forms
+
+
+class EmailPostForm(forms.Form):
+    name = forms.CharField(max_length=25)
+    email = forms.EmailField()
+    to = forms.EmailField()
+    comments = forms.CharField(required=False, widget=forms.Textarea)
```
Comment (waranlogesh) [inline mysite/blog/forms.py]: This name is ambiguous since the post will also be having comments. We can rename it as share_text.

File: mysite/blog/views.py [hunk 9]
```diff
@@ -21,3 +24,24 @@ def post_detail(request, year, month, day, post):
...
+                form.cleaned_data["to"],
+            )
+            messages.success(
+                request, f"{post.title} was successfully sent to {form.to}"
+            )
+        else:
+            messages.error(request, "Error sending mail")
+    else:
+        form = EmailPostForm()
+    return render(request, "blog/post/share.html", {"post": post, "form": form})
```
Comment (waranlogesh) [inline mysite/blog/views.py]: These branching conditionals can be simplified to
```suggestion
    if request.method == "POST":
        form = EmailPostForm(request.POST)
        if form.is_valid():
            share_post_by_email(
                post,
                form.cleaned_data["name"],
                form.cleaned_data["comments"],
                form.cleaned_data["to"],
            )
            messages.success(
                request, f"{post.title} was successfully sent to {form.to}"
            )
            return render(request, "blog/post/share.html", {"post": post, "form": form})
            
        messages.error(request, "Error sending mail")
        return render(request, "blog/post/share.html", {"post": post, "form": form})
 
    form =  EmailPostForm()
    return render(request, "blog/post/share.html", {"post": post, "form": form})
```

Too many branching conditionals makes the code difficult to read and complex.

File: mysite/mysite/settings.py [hunk 10]
```diff
@@ -119,3 +119,10 @@
 # https://docs.djangoproject.com/en/3.0/howto/static-files/
 
 STATIC_URL = "/static/"
+
+EMAIL_BACKEND = "django.core.mail.backends.console.EmailBackend"
+EMAIL_HOST = "smtp.gmail.com"
+EMAIL_HOST_USER = "your_account@gmail.com"
+EMAIL_HOST_PASSWORD = "your_password"
+EMAIL_PORT = 587
+EMAIL_USE_TLS = True
```
Comment (waranlogesh) [inline mysite/mysite/settings.py]: Add https://django-environ.readthedocs.io/en/latest/ in master and use it.

**Review by waranlogesh — CHANGES_REQUESTED** [review]

**Review by waranlogesh — APPROVED** [review]
